exporters/committime/collector_base.py

```diff
@@ -236,16 +235,26 @@ def __parse_repourl(self):
         """Parse the repo_url into individual pieces"""
         if self.__repo_url is None:
             return
-
-        url_tokens = self.__repo_url.split("/")
-        self.__repo_protocol = url_tokens[0]
-        if self.__repo_protocol.endswith(':'):
-            self.__repo_protocol = self.__repo_protocol[:-1]
-        # token 1 is always a blank
-        self.__repo_fqdn = url_tokens[2]
-        self.__repo_group = url_tokens[3]
-        self.__repo_name = url_tokens[4]
-        if self.__repo_name.endswith('.git'):
-            self.__repo_project = self.__repo_name[:-4]
+        parsed = giturlparse.parse(self.__repo_url)
+        # In the case of multiple subgroups the host will be in the pathname
+        # Otherwise, it will be in the resource
+        if parsed.pathname.startswith('//'):
+            self.__repo_fqdn = parsed.pathname.split("/")[2]
+        else:
+            self.__repo_fqdn = parsed.resource
+        # This ugly block of code is due to the regex handling of the protocol in git-url-parse package
+        # In cases where the protocol is found to be 'git', 'http(s)' or 'ssh' it is handled fine
+        # but is not graceful for strange urls like 'kmoos://myhost/mygroup/repo.git'
+        if parsed.protocol == 'ssh' and len(parsed.protocols) > 0:
+            if parsed.protocols[0] == 'https':
+                self.__repo_protocol = parsed.protocols[0]
+            elif parsed.protocols[0] not in ['http', 'https', 'git', 'ssh']:
+                self.__repo_protocol = parsed.protocol
+                self.__repo_fqdn = self.__repo_url.split('/')[2]
+            else:
+                self.__repo_protocol = parsed.protocol
```

Comment:
Ah, I didn't realize until this comment that you were doing all of this to make the `kmoos://...` test work. I think the actual answer here is that git is more prescriptive than we thought it was, and this is not a valid test. Originally we were really just check of the format of `<something>://<something>/<bla>`, which clearly wasn't sufficient. Let's remove the tests that don't make sense, and remove this code. Sorry for the run around.

Reply:
I'm actually relieved to hear that is not proper grammar.  No problems backing out this code.